### PR TITLE
Cover wallet send events with tests Part 2 #20411 #20533

### LINKED ISSUE #20411: Cover wallet send events with tests
# Problem

Events in `status_im/context/wallet/send/events.cljs` aren't covered with tests. 
Since this is the most crucial part of wallet, it should be properly tested.

## Implementation

These events should be covered:

- [x] :wallet/clean-send-data (Milad)
- [x]       :wallet/select-address-tab (Milad)
- [x]       :wallet/suggested-routes-success (Milad)
- [x]      :wallet/suggested-routes-error (Milad)
- [x]      :wallet/clean-suggested-routes (Milad)
- [x]      :wallet/clean-send-address (Milad)
- [x]      :wallet/clean-send-amount (Milad)
- [x]      :wallet/clean-disabled-from-networks (Milad)
- [x]      :wallet/clean-from-locked-amounts (Milad)
- [x]      :wallet/select-send-address (Milad)
- [x]      :wallet/update-receiver-networks (Volodymyr)
- [x]      :wallet/set-token-to-send (Volodymyr)
- [x]      :wallet/edit-token-to-send (Volodymyr)
- [x]      :wallet/clean-selected-token (Milad)
- [x]      :wallet/clean-selected-collectible (Milad)
- [x]      :wallet/set-collectible-to-send (Volodymyr)
- [x]      :wallet/set-collectible-amount-to-send (Volodymyr)
- [x]      :wallet/set-token-amount-to-send (Volodymyr)
- [x]      :wallet/disable-from-networks (Milad)
- [x]      :wallet/lock-from-amount (Milad)
- [x]      :wallet/unlock-from-amount (Milad)
- [x]      :wallet/reset-network-amounts-to-zero (Milad)
- [ ]      :wallet/get-suggested-routes (Milad)
- [x]      :wallet/add-authorized-transaction (Milad)
- [ ]      :wallet/close-transaction-progress-page (Milad)
- [ ]      :wallet/send-transaction (Milad)
- [x]      :wallet/select-from-account (Milad)

## Acceptance Criteria

Tests written for all the important events

### PR DESCRIPTION
**Pull request by mmilad75**
fixes https://github.com/status-im/status-mobile/issues/20411

Part 1: https://github.com/status-im/status-mobile/pull/20533

This PR adds the rest of the tests to the wallet send events.

### CODE REVIEW

File: src/status_im/contexts/wallet/send/events_test.cljs
Comment (ilmotta) [inline src/status_im/contexts/wallet/send/events_test.cljs]: We could clean up a few assertions and be more direct, more readable. Possibly in a few other tests you could use the same idea. If we can get away with a simple equality check the test is potentially more robust. For example, the original test still passes even if the event handler changes other parts of the app-db, but this would clearly be a bug given its name/purpose (clean send address).

It could be simply:

```clojure
(h/deftest-event :wallet/clean-send-address
  [event-id dispatch]
  (let [initial-db])
  (reset! rf-db/app-db
    {:wallet {:ui {:send {:to-address  "0x01"
                          :recipient   {:recipient-type :saved-address
                                        :label          "label"}
                          :other-props :value}}}})
  (is (= {:db {:wallet {:ui {:send {:other-props value}}}}}
         (dispatch [event-id]))))
```

In general, I think the event test can catch more and be more useful if we verify the entire output of the handler. The moment we use `get-in` and grab only slices of the map of effects, we are allowing the implementation to change without coverage. Sometimes it makes sense, but here and in some other tests I think it's not beneficial.

File: src/status_im/contexts/wallet/send/events_test.cljs
Comment (ilmotta) [inline src/status_im/contexts/wallet/send/events_test.cljs]: This map is almost unreadable because it's too right-aligned.

File: src/status_im/contexts/wallet/send/events_test.cljs
Comment (ilmotta) [inline src/status_im/contexts/wallet/send/events_test.cljs]: Here the map is also difficult to read because it's too right-aligned, which forces zprint to add many undesirable line breaks (something a human wouldn't do I mean).

**Review by ilmotta — COMMENTED** [review]
It's a pretty impressive amount of tests we got here. Thank you @mmilad75.

I think a handful of tests could be cleaned-up a little bit, they have a lot of repetition with `get-in`, `contains?`, etc that could be replaced with simple equality checks (I made a separate comment with an example).

File: src/status_im/contexts/wallet/send/events_test.cljs
Comment (J-Son89) [inline src/status_im/contexts/wallet/send/events_test.cljs:219]: @VolodLytvynenko - this test is not needed any more right?

File: src/status_im/contexts/wallet/send/events_test.cljs
Comment (J-Son89) [inline src/status_im/contexts/wallet/send/events_test.cljs]: @VolodLytvynenko - this test is not needed any more right?

File: src/status_im/contexts/wallet/send/events_test.cljs
Comment (J-Son89) [inline src/status_im/contexts/wallet/send/events_test.cljs]: @VolodLytvynenko - this test is not needed any more right?

**Review by J-Son89 — APPROVED** [review]
thank you for adding these tests @mmilad75!

File: src/status_im/contexts/wallet/send/events_test.cljs
Comment (VolodLytvynenko) [reply src/status_im/contexts/wallet/send/events_test.cljs]: it seems @vkjr this is addressed to you.

File: src/status_im/contexts/wallet/send/events_test.cljs
Comment (VolodLytvynenko) [reply src/status_im/contexts/wallet/send/events_test.cljs]: it seems @vkjr this is addressed to you.

File: src/status_im/contexts/wallet/send/events_test.cljs
Comment (J-Son89) [reply src/status_im/contexts/wallet/send/events_test.cljs]: Sorry! @vkjr was the intention 😅

File: src/status_im/contexts/wallet/send/events_test.cljs
Comment (mmilad75) [reply src/status_im/contexts/wallet/send/events_test.cljs]: But you're using `=` in the assertion. Is that ok?

File: src/status_im/contexts/wallet/send/events_test.cljs
Comment (mmilad75) [reply src/status_im/contexts/wallet/send/events_test.cljs]: @J-Son89 Why not?

**Comment by vkjr** [comment]
@J-Son89, tests for locked amounts aren't really relevant now. But they are testing the code that still exists in the codebase. So I suggest considering tests+code as a "bundle" :) If we decide to remove that code from the codebase - let's do along with tests. But I'd prefer not to remove the tests alone because that would leave the code untested and while it is still there, it would be better to have it tested, wdyt?

File: src/status_im/contexts/wallet/send/events_test.cljs
Comment (vkjr) [reply src/status_im/contexts/wallet/send/events_test.cljs]: @mmilad75, we disabled the functionality for locking amounts so these events won't be used in the code

File: src/status_im/contexts/wallet/send/events_test.cljs
Comment (ilmotta) [reply src/status_im/contexts/wallet/send/events_test.cljs]: FYI: Just opened PR https://github.com/status-im/status-mobile/pull/20825.

**Comment by J-Son89** [comment]
> @J-Son89, tests for locked amounts aren't really relevant now. But they are testing the code that still exists in the codebase. So I suggest considering tests+code as a "bundle" :) If we decide to remove that code from the codebase - let's do along with tests. But I'd prefer not to remove the tests alone because that would leave the code untested and while it is still there, it would be better to have it tested, wdyt?

Yeah I agree, very good point! Sounds good to me!

File: src/status_im/contexts/wallet/send/events_test.cljs
Comment (ilmotta) [inline src/status_im/contexts/wallet/send/events_test.cljs]: I think we should at least remove the obvious duplication from the tests in this file. There's no need to repeatedly `dispatch` and `get-in` the same thing. Like so:

```clojure
(h/deftest-event :wallet/clean-selected-collectible
  [event-id dispatch]
  (reset! rf-db/app-db
    {:wallet {:ui {:send {:other-props        :value
                          :collectible        "ETH"
                          :token-display-name "ETH"
                          :amount             10}}}})
  (let [send (get-in (dispatch [event-id]) [:db :wallet :ui :send])]
    (is (not (contains? send :collectible)))
    (is (not (contains? send :token-display-name)))
    (is (not (contains? send :amount)))
    (is (contains? send :other-props))))
```

Better yet I think would be to replace all these assertions with only one that checks the output of the function, but for that to work you would need to use this branch which I just opened https://github.com/status-im/status-mobile/pull/20825, but up to you if you want to wait until that one is merged, probably better to get this PR in `develop` first.

```clojure
(h/deftest-event :wallet/clean-selected-collectible
  [event-id dispatch]
  (reset! rf-db/app-db
    {:wallet {:ui {:send {:other-props        :value
                          :collectible        "ETH"
                          :token-display-name "ETH"
                          :amount             10}}}})
  (is (match-strict? {:db {:wallet {:ui {:send {:other-props :value}}}}}
                     (dispatch [event-id]))))
```

File: src/status_im/contexts/wallet/send/events_test.cljs
Comment (ilmotta) [inline src/status_im/contexts/wallet/send/events_test.cljs]: Lovely `tap>` leftover :)

**Review by ilmotta — APPROVED** [review]

File: src/status_im/contexts/wallet/send/events_test.cljs
Comment (mmilad75) [reply src/status_im/contexts/wallet/send/events_test.cljs]: Thanks for noticing :D. Removed

**Pull request merged**
